hadoop-hdds/docs/content/feature/Nonrolling-Upgrade.md

```diff
@@ -0,0 +1,110 @@
+    - The cluster will remain fully operational with all functionality present in the old version still allowed.
+    - Any data created while pre-finalized will remain readable after downgrade.
+
+2. **Finalized**: When a finalize command is given to OM or SCM, they will enter a finalized state.  In the finalized state:
+    - The cluster can no longer be downgraded.
+    - All new features of the cluster introduced in the new version can be used.
+
+### Querying finalization status
+
+**OM**: `ozone admin om finalizationstatus`. If using OM HA, finalization status is checked for the quorum, not individual OMs.
+
+**SCM**: `ozone admin scm finalizationstatus`. SCM will report that finalization is complete once it has finalized and is aware of enough finalized datanodes to form a write pipeline. The remaining datanodes will finalize asynchronously and be incorporated into write pipelines after informing SCM that they have finalized.
+
+**Datanodes**: `ozone admin datanode list` will list all datanodes and their health state as seen by SCM. If SCM is finalized, then datanodes whose health state is `HEALTHY` have informed SCM that they have finalized. Datanodes whose health state is `HEALTHY_READONLY` have not yet informed SCM that they have finished finalization. `HEALTHY_READONLY` (pre-finalized) datanodes remain readable, so the cluster is operational even if some otherwise healthy datanodes have not yet finalized. `STALE` or `DEAD` datanodes will be told to finalize by SCM once they are reachable again.
+
+
+## Steps to upgrade and downgrade
+
+Starting with your current version of Ozone, complete the following steps to upgrade to a newer version of Ozone.
+
+1. If using OM HA and currently running Ozone 1.2.0 or greater, prepare the Ozone Manager.
+    ```
+    ozone admin om -id=<om-sevice-id> prepare
+    ```
+    If OM HA is not being used, this step can be skipped. This will block the Ozone Managers from receiving all write requests. See [Ozone Manager Prepare For Upgrade]({{< relref "../design/omprepare.md" >}}) for more information
+
+2.  Stop all components.
+
+3. Replace artifacts of all components with the newer versions.
+
+4. Start the components
+    1. Start the SCM and datanodes as usual:
+        ```
+        ozone --daemon scm start
+        ```
+        ```
+        ozone --daemon datanode start
+        ```
+
+    2. Start the Ozone Manager using the `--upgrade` flag to take it out of prepare mode.
+        ```
+        ozone --daemon om start --upgrade
+        ```
+        - There also exists a `--downgrade` flag which is an alias of `--upgrade`. The name used does not matter.
+
+        - **IMPORTANT**: All OMs must be started with the `--upgrade`  or `--downgrade` flag in this step. If some or none of the OMs are started with this flag by mistake, run `ozone admin om -id=<om-sevice-id> cancelprepare` to make sure all OMs leave prepare mode.
```

Comment:
This phrase seems weird: " If some or none of the OMs".  Should that just be: "If some of the OMs"?

Reply:
That's right. I will update it to just say 'some of the OMs'.